### PR TITLE
Versioning Resources

### PR DESCRIPTION
**Pull request by jerop**
# Changes

Today, we don't always bump versions of resources when updating them. The guideline has been to bump versions only when behavior changes, but it's hard to figure out when the behavior has changed (a change that could be trivial to one user could be meaningful to another).

Not bumping resource versions when changing them causes issues where the resource definition becomes dependent on the time when it was applied by the user - which causes unexpected failures as described in https://github.com/tektoncd/catalog/issues/784.

This issue also came up as an issue where users cannot depend on the Step indices because they can change: https://github.com/tektoncd/community/pull/572#pullrequestreview-836578705.

In TEP-0003, we already proposed that a policy for versioning of resources:
https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md#versioning-resources

In Catalog Working Group on 01/13/2022, we revisited that policy and:
- agreed to follow the versioning policy
- make it easier to bump resources (see https://github.com/tektoncd/plumbing/pull/994 - thanks @sbwsg 🎉)

This change documents the versioning policy in the contributions guide.

Fixes https://github.com/tektoncd/catalog/issues/784

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [n/a] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

**Review by vinamra28 — APPROVED** [review]
/cc @tektoncd/catalog-maintainers

**Comment by chmouel** [comment]
if every time there is an update of a software which is in a task or we have to fix something in it (like documentation) we have to bump this may results a large number of tasks. 

It's maybe find and I am not sure if it's bad or not  and could end up with some scaling issues.

I wish we could use git as versioning system instead of directory structure but I am not sure how we can achieve that.

so basically  my vote is neutral :)

**Comment by jerop** [comment]
hoping the work in [TEP-0101](https://github.com/tektoncd/community/pull/663) will address this issue, so closing this pull request
